src/pydata_sphinx_theme/assets/scripts/index.js

```diff
@@ -71,7 +79,74 @@ function scrollToActive() {
   });
 }
 
+////////////////////////////////////////////////////////////////////////////////
+// Theme interaction
+////////////////////////////////////////////////////////////////////////////////
+var prefersDark = window.matchMedia("(prefers-color-scheme: dark)");
+
+function autoTheme(e) {
+  document.body.dataset.theme = prefersDark.matches ? "dark" : "light";
+}
+
+function setTheme(mode) {
+  if (mode !== "light" && mode !== "dark" && mode !== "auto") {
+    console.error(`Got invalid theme mode: ${mode}. Resetting to auto.`);
+    mode = "auto";
+  }
+
+  // get the theme
+  var colorScheme = prefersDark.matches ? "dark" : "light";
+  document.body.dataset.mode = mode;
+  document.body.dataset.theme = mode == "auto" ? colorScheme : mode;
+
+  // save mode
+  localStorage.setItem("theme", mode);
+  console.log(`Changed to ${mode} mode.`);
+
+  // add a listener if set on auto
+  prefersDark.onchange = mode == "auto" ? autoTheme : "";
+}
+
+function cycleTheme() {
```

Comment:
Please provide docstrings for each of the functions here (use [typescript comment styles](https://marketplace.visualstudio.com/items?itemName=salbert.comment-ts) if you like)